### DISCUSSION TITLE
How to use arguments from command line?

### DISCUSSION THREAD

**Discussion by pedroripper** [FilledStacks/stacked#1089] [Q&A]
Hi

I am using Stacked to build a desktop application that should be able to open a `.sqlite` file from the command line using its release (example below).

```
> my_app.exe "C:\path_to_file\file.sqlite
```

If no arguments are passed, it goes to a view where the user can open the file explorer to select the file themself. However, I am having trouble to use the path given as argument in the command line to skip the selecting file part.

Currently I am able to get the path argument passing it as a parameter for the `main` function.

```dart
Future<void> main(List<String> args) async {
  WidgetsFlutterBinding.ensureInitialized();
  await ThemeManager.initialise();
  setPathUrlStrategy();
  await setupLocator(stackedRouter: stackedRouter);
  setupDialogUi();
  setupBottomSheetUi();

  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  // ...

  @override
  Widget build(BuildContext context) {
    return ThemeBuilder(
      defaultThemeMode: ThemeMode.dark,
      lightTheme: buildLightTheme(),
      darkTheme: buildDarkTheme(),
      builder: (context, regularTheme, darkTheme, themeMode) => ResponsiveApp(
        preferDesktop: true,
        builder: (_) => MaterialApp.router(
          theme: regularTheme,
          darkTheme: darkTheme,
          themeMode: themeMode,
          debugShowCheckedModeBanner: false,
          localizationsDelegates: AppLocalizations.localizationsDelegates,
          supportedLocales: AppLocalizations.supportedLocales,
          routerDelegate: stackedRouter.delegate(),
          routeInformationParser: stackedRouter.defaultRouteParser(),
        ),
      ),
    );
  }
}
```

What I would like to know is if there is a simple way to pass this path to a specific view using methods from Stacked.

We have a Service for opening the database for a given path

```dart
class DatabaseService with ListenableServiceMixin {
  String databasePath = "";

  Future<void> init() async {}
}
```

And this Service is initialized in the SplashView

```dart 
class SplashViewModel extends BaseViewModel {
  final _databaseService = locator<DatabaseService>();
  final _routerService = locator<RouterService>();

  Future<void> initialize() async {
    await _databaseService.init();;

    await _routerService.replaceWith(const StartupViewRoute());
  }
}
```

Thanks in advance!

**Comment by FilledStacks** [comment]
Hey, 

There's no first-party support for this, but what I would do is take my database service registration out of the `app.dart` file and then have a `customLocatorSetup` function that I can pass args too. 

this way you can register your database service with the args passed into main.
